services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/FlowValidationNegativeSpec.groovy

```diff
@@ -161,4 +166,44 @@ class FlowValidationNegativeSpec extends BaseSpecification {
         def switches = topology.getActiveSwitches()
         return [switches.first(), switches.first()]
     }
+
+    @Unroll
+    def "Unable to #action flow on a non-existent flow"() {
+        when: "Try to #action rules on a non-existent flow"
+        northbound."$method"(NON_EXISTENT_FLOW_ID)
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Could not reroute flow: Flow $NON_EXISTENT_FLOW_ID not found"
+
+        where:
+        action        | method
+        "synchronize" | "synchronizeFlow"
+        "reroute"     | "rerouteFlow"
+    }
+
+    def "Unable to get flow on a non-existent flow"() {
+        when: "Try to get rules on a non-existent flow"
+        northbound.getFlow(NON_EXISTENT_FLOW_ID)
+
+        then: "An error is received (404 code)"
+        def exc = thrown(HttpClientErrorException)
+        exc.rawStatusCode == 404
+        exc.responseBodyAsString.to(MessageError).errorMessage ==
+                "Can not get flow: Flow $NON_EXISTENT_FLOW_ID not found"
+    }
+
+    @Ignore("https://github.com/telstra/open-kilda/issues/2027")
+    def "Unable to validate flow on a non-existent flow"() {
+        when: "Try to validate rules on a non-existent flow"
```

Comment:
`Try to validate non-existent flow`/`Try calling 'validate flow' with non-existent flow id`
1. You can validate rules 'on switch', 'on all switches of the flow', but not 'on flow' itself.
2. It will potentially validate not only rules, bot also meters
3. It creates confusion between 'switch rules validation' and 'flow validation', which are different operations